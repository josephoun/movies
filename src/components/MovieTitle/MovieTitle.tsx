import React from "react";
import {MovieType} from "../../types/MovieType";

type MovieItemPropsType = {
    movie: MovieType;
}
export default function MovieTitle({movie} : MovieItemPropsType) {
    return (
        <>
             {
                 movie.title ? (
                     movie.title + ( movie.released ? ' (' + movie.released + ')' : '')
                 ) : 'Untitled'
             }
        </>
    )
}

