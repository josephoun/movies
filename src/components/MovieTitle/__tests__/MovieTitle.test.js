import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom'
import MovieTitle from "../MovieTitle";

describe("MovieTitle component", () => {

    it("should match snapshot with title and released", () => {
        const { container } = render(
            <BrowserRouter>
                <MovieTitle movie={{title: 'FAKE_TITLE', released: 'FAKE_RELEASED'}} />
            </BrowserRouter>,
        );
        expect(container).toMatchSnapshot();
    });

    it("should match snapshot with title and without released", () => {
        const { container } = render(
            <BrowserRouter>
                <MovieTitle movie={{title: 'FAKE_TITLE'}} />
            </BrowserRouter>,
        );
        expect(container).toMatchSnapshot();
    });

    it("should match snapshot without title and without released", () => {
        const { container } = render(
            <BrowserRouter>
                <MovieTitle movie={{}} />
            </BrowserRouter>,
        );
        expect(container).toMatchSnapshot();
    });
});
