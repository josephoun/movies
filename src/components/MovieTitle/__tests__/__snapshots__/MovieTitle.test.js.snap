// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitle component should match snapshot with title and released 1`] = `
<div>
  FAKE_TITLE (FAKE_RELEASED)
</div>
`;

exports[`MovieTitle component should match snapshot with title and without released 1`] = `
<div>
  FAKE_TITLE
</div>
`;

exports[`MovieTitle component should match snapshot without title and without released 1`] = `
<div>
  Untitled
</div>
`;
